tailscale/provider.go

```diff
@@ -286,3 +288,14 @@ func readWithWaitFor(fn schema.ReadContextFunc) schema.ReadContextFunc {
 		}
 	}
 }
+
+// setProperties sets the properties of a ResourceData using the values in a
+// map.
+func setProperties(d *schema.ResourceData, props map[string]any) diag.Diagnostics {
```

Comment:
```suggestion
// setProperties sets the provided map props entries as properties of a ResourceData.
func setProperties(d *schema.ResourceData, props map[string]any) diag.Diagnostics {
```
take or leave, trying to make it a tad clearer that it doesn't clear out any existing properties, it's purely additive